pallets/delegation/src/lib.rs

```diff
@@ -169,53 +186,46 @@ decl_module! {
 		) -> DispatchResult {
 			// origin of the transaction needs to be a signed sender account
 			let sender = ensure_signed(origin)?;
+
 			// check if a delegation node with the given identifier already exists
-			if <Delegations<T>>::contains_key(delegation_id) {
-				return Self::error(Self::ERROR_DELEGATION_ALREADY_EXISTS);
-			}
+			ensure!(!<Delegations<T>>::contains_key(delegation_id), Error::<T>::AlreadyExists);
+
 			// calculate the hash root and check if the signature matches
 			let hash_root = Self::calculate_hash(delegation_id, root_id, parent_id, permissions);
-			if !verify_encoded_lazy(&delegate_signature, &&hash_root, &delegate) {
-				return Self::error(Self::ERROR_BAD_DELEGATION_SIGNATURE);
-			}
+			ensure!(verify_encoded_lazy(&delegate_signature, &&hash_root, &delegate), Error::<T>::BadSignature);
 
 			// check if root exists
-			let root = <error::Module<T>>::ok_or_deposit_err(
-				<Root<T>>::get(root_id),
-				Self::ERROR_ROOT_NOT_FOUND
-			)?;
+			let root = <Root<T>>::get(root_id).ok_or(Error::<T>::RootNotFound)?;
+
 			// check if this delegation has a parent
 			if let Some(parent_id) = parent_id {
 				// check if the parent exists
-				let parent_node = <error::Module<T>>::ok_or_deposit_err(
-					<Delegations<T>>::get(parent_id),
-					Self::ERROR_PARENT_NOT_FOUND
-				)?;
+				let parent_node = <Delegations<T>>::get(parent_id).ok_or(Error::<T>::ParentNotFound)?;
+
 				// check if the parent's delegate is the sender of this transaction and has permission to delegate
-				if !parent_node.owner.eq(&sender) {
-					return Self::error(Self::ERROR_NOT_OWNER_OF_PARENT);
-				} else if (parent_node.permissions & Permissions::DELEGATE) != Permissions::DELEGATE {
-					return Self::error(Self::ERROR_NOT_AUTHORIZED_TO_DELEGATE);
-				} else {
-					// insert delegation
-					debug::print!("insert Delegation with parent");
-					<Delegations<T>>::insert(delegation_id, DelegationNode::<T>::new_child(
-						root_id,
-						parent_id,
-						delegate.clone(),
-						permissions,
-					));
-					// add child to tree structure
-					Self::add_child(delegation_id, parent_id);
-				}
+				ensure!(parent_node.owner.eq(&sender), Error::<T>::NotOwnerOfParent);
+
+				//
```

Comment:
I guess that wanted to become a comment? 😁

Reply:
True! Fixed in 19510b 🙃